portal/src/GTMProvider.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { useMemo } from "react";
+import {
+  GTMProvider as ReactHookGTMProvider,
+  useGTMDispatch as useReactHookGTMDispatch,
+} from "@elgorditosalsero/react-gtm-hook";
+import { useAppContext } from "./context/AppContext";
+
+export type TrackValue = string | string[] | boolean | undefined;
+export type EventData = Record<string, TrackValue>;
+
+export interface AuthgearGTMEvent {
+  event: AuthgearGTMEventType;
+  app_id?: string;
+  event_data?: EventData;
+  _clear: boolean;
+}
+
+export enum AuthgearGTMEventType {
+  CreatedProject = "ag.event.createdProject",
+  ClickedGetStarted = "ag.event.clickedGetStarted",
+  CreatedApplication = "ag.event.createdApplication",
+  InvitedAdmin = "ag.event.invitedAdmin",
+  AddedSSOProviders = "ag.event.addedSSOProviders",
+  ClickedDocLink = "ag.event.clickedDocLink",
+  ClickedNextInProjectWizard = "ag.event.clickedNextInProjectWizard",
+  ClickedSkipInProjectWizard = "ag.event.clickedSkipInProjectWizard",
+}
+
+interface AuthgearGTMEventParams {
+  event: AuthgearGTMEventType;
+  eventData?: EventData;
+}
+
+export function useAuthgearGTMEvent({
```

Comment:
Instead of requiring the caller to provide a full event, I suggest we return an object that can aid constructing a full event. The benefit is that if we need to send multiple events, we need not call the hook multiple times.

```diff
diff --git a/portal/src/GTMProvider.tsx b/portal/src/GTMProvider.tsx
index 62ecd58bf..ec0ad5ca9 100644
--- a/portal/src/GTMProvider.tsx
+++ b/portal/src/GTMProvider.tsx
@@ -8,13 +8,16 @@ import { useAppContext } from "./context/AppContext";
 export type TrackValue = string | string[] | boolean | undefined;
 export type EventData = Record<string, TrackValue>;
 
-export interface AuthgearGTMEvent {
-  event: AuthgearGTMEventType;
+export interface AuthgearGTMEventBase {
   app_id?: string;
-  event_data?: EventData;
   _clear: boolean;
 }
 
+export interface AuthgearGTMEvent extends AuthgearGTMEventBase {
+  event: AuthgearGTMEventType;
+  event_data?: EventData;
+}
+
 export enum AuthgearGTMEventType {
   CreatedProject = "ag.event.createdProject",
   ClickedGetStarted = "ag.event.clickedGetStarted",
@@ -26,15 +29,7 @@ export enum AuthgearGTMEventType {
   ClickedSkipInProjectWizard = "ag.event.clickedSkipInProjectWizard",
 }
 
-interface AuthgearGTMEventParams {
-  event: AuthgearGTMEventType;
-  eventData?: EventData;
-}
-
-export function useAuthgearGTMEvent({
-  event,
-  eventData,
-}: AuthgearGTMEventParams): AuthgearGTMEvent {
+export function useAuthgearGTMEventBase(): AuthgearGTMEventBase {
   let appContextID: string | undefined;
   try {
     const appContext = useAppContext();
@@ -43,16 +38,27 @@ export function useAuthgearGTMEvent({
 
   return useMemo(() => {
     return {
-      event: event,
       app_id: appContextID,
-      event_data: eventData,
       // Prevent GTM recursive merge event data object
       // https://github.com/google/data-layer-helper#preventing-default-recursive-merge
       _clear: true,
     };
-  }, [event, eventData, appContextID]);
+  }, [appContextID]);
 }
 
+// Usage
+// const send = useGTMDispatch();
+// const base = useAuthgearGTMEventBase();
+//
+// send({
+//    ...base,
+//    event: AuthgearGTMEventType.CreatedProject,
+// });
+// send({
+//    ...base,
+//    event: ClickedGetStarted,
+// });
+
 interface AuthgearGTMEventDataAttributesParams {
   event: AuthgearGTMEventType;
   eventDataAttributes?: Record<string, string>;

```